src/absil/ilwrite.fsi

```diff
@@ -15,17 +15,22 @@ type ILStrongNameSigner =
     static member OpenKeyPairFile: string -> ILStrongNameSigner
     static member OpenKeyContainer: string -> ILStrongNameSigner
 
+type EmitStreamProvider = Lazy<Choice<EmitTo, Diagnostic>>
```

Comment:
I find using ``Lazy`` here is a bit awkward - in general it's just rarely a good type to expose in APIs.  I think I'd prefer just 

    [<RequireQualifiedAccess>]
    type EmitTo = 
        | File of string
        | Stream of (unit -> Stream)

    type options = 
        { .. pdbfile : EmitTo option ...  }    

Internally to ilwrite.fs you can convert to a register that amortizes the evaluation in the EmitTo case.  If tracking diagnostics is a problem then I'm OK with you using 

        | Stream of (unit -> Choice<Stream, exn>)

or

        | Stream of (unit -> Choice<Stream, Diagnostic>)

Could you see if that works?